base/cvd/cuttlefish/host/commands/cvd/server_client.cpp

```diff
@@ -81,30 +82,44 @@ Result<void> SendResponse(const SharedFD& client,
   return {};
 }
 
-RequestWithStdio::RequestWithStdio(cvd::Request message,
-                                   std::vector<SharedFD> fds)
-    : message_(message), fds_(std::move(fds)) {}
-
-const cvd::Request& RequestWithStdio::Message() const { return message_; }
+RequestWithStdio RequestWithStdio::StdIo(cvd::Request message) {
+  return RequestWithStdio(std::move(message), std::cin, std::cout, std::cerr);
+}
 
-const std::vector<SharedFD>& RequestWithStdio::FileDescriptors() const {
-  return fds_;
+static std::istream& NullIn() {
+  static std::ifstream* in = new std::ifstream("/dev/null");
+  return *in;
 }
 
-SharedFD RequestWithStdio::In() const {
-  return fds_.size() > 0 ? fds_[0] : SharedFD();
+static std::ostream& NullOut() {
+  static std::ofstream* out = new std::ofstream("/dev/null");
+  return *out;
 }
 
-SharedFD RequestWithStdio::Out() const {
-  return fds_.size() > 1 ? fds_[1] : SharedFD();
+RequestWithStdio RequestWithStdio::NullIo(cvd::Request message) {
+  return RequestWithStdio(std::move(message), NullIn(), NullOut(), NullOut());
 }
 
-SharedFD RequestWithStdio::Err() const {
-  return fds_.size() > 2 ? fds_[2] : SharedFD();
+RequestWithStdio RequestWithStdio::InheritIo(cvd::Request message,
```

Comment:
nit: `InheritIO` to match the style of `SharedFD` (instead of `InheritIo` and `SharedFd`)

Reply:
The [style guide](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules) writes

> For names written in mixed case (also sometimes referred to as "[camel case](https://en.wikipedia.org/wiki/Camel_case)" or "[Pascal case](https://en.wiktionary.org/wiki/Pascal_case)"), in which the first letter of each word is capitalized, prefer to capitalize abbreviations as single words, e.g., `StartRpc()` rather than `StartRPC()`